### ISSUE TITLE
Board Design

### ISSUE THREAD

**Issue by mrsstaceyseigle** [magkopian/raspberry-pi-compute-module-base-board#2]
Could you put together a board for me that only has 1 usb port, and power. Also could you help with integrating Ethernet port and wifi without using usb dongle

**Comment by boulnat** [comment]
Hi, let me have a look at your request.
Do you need to be able to switch between Ethernet and WiFi or do you need both at the same time ?
I guess you will need hdmi to configure your board ?
What do you mean by USB port and power ?

**Comment by magkopian** [comment]
@mrsstaceyseigle I'm not quite sure what you mean by "1 usb port, and power". The board already has one USB Type A connector for attaching USB devices and a micro USB used for power.

As for adding Ethernet and WiFi to the board that will depend on your requirements. For example, if you are fine with 10Mbit Ethernet probably the easiest way is to use an [ENC28J60](http://ww1.microchip.com/downloads/en/DeviceDoc/39662e.pdf) as it doesn't require a lot of external components and uses the SPI bus. You can have a look on page 7 of the datasheet for an application diagram.

I also designed a microcontroller board based on it in the past. That was though a long time ago and was done on Eagle but [here](https://www.instructables.com/id/PIC18-Development-Board-with-Ethernet-and-USB/) is the project if you want to have a look. The PCB design files are attached at the end of step 2. Regarding the Pi specifically I've never attempted to connect an ENC28J60 to it via SPI, but I do know a driver for this chip already exists in the Linux kernel for a while now and quite a few people where successful in doing so.

As for WiFi, probably the easiest way is to use an ESP8266 module as it should require virtually no external components. That's not something I've done myself, but I'm aware of people using ESP8266 modules to [add WiFi](http://pwiatrowski.com/technology/raspberry-pi-zero-esp8266-internet/) to Raspberry Pi Zero based projects before the release of the W variant which features WiFi.

**Comment by mrsstaceyseigle** [comment]
I was hoping to make a board like in the photo. Ethernet could be 10Mbit as all will be done is transfering text data very fast. Mainly looking at stable connection with Ethernet. Wifi could be 10Mbit too as long as there are not connection issues with newer routers. 

My question also would be lets say that we went with a wifi module or Ethernet Module that required more components would they connect to the same points (bus) as the units your mention. 

magkopian and boulnat could you put together a board layout like shown in this photo. A very narrow and small board that is a little bigger than the raspberry pi compute module on requested components 
![board layout](https://user-images.githubusercontent.com/43449594/74443610-7de19d00-4e41-11ea-881e-ba81283e30cd.png)

**Comment by magkopian** [comment]
> Ethernet could be 10Mbit as all will be done is transfering text data very fast. Mainly looking at stable connection with Ethernet. Wifi could be 10Mbit too as long as there are not connection issues with newer routers.

If all you need to do is transfer text data, then an ENC28J60 Ethernet controller as well as an ESP8266 WiFi module should be both perfectly fine for your needs.

> My question also would be lets say that we went with a wifi module or Ethernet Module that required more components would they connect to the same points (bus) as the units your mention.

Depends on the chip, the ENC28J60 for example connects to the SPI bus but there are other ICs like for example the LAN9512 used by the original Raspberry Pi Model B that connect to the USB bus.

The LAN9512 also functions as a 2-port USB hub and with it you get 100Mbit Ethernet. But it's harder to use compared to the ENC28J60, as it requires more external components and is only available as a QFN package as far as I'm aware which makes it harder to solder. That is though the chip I'd personally use if I wanted to add Ethernet to a CM based project, unless I knew for a fact that I won't be needing 100Mbit Ethernet and wanted to drop the production cost as much as possible, as the cost of just the LAN9512 is more or less double that of an ENC28J60.

> could you put together a board layout like shown in this photo. A very narrow and small board that is a little bigger than the raspberry pi compute module on requested components

If your question is whether I can design a board like this, then the answer is certainly. However, that doesn't mean I also have the time to work on something like that at this point in time. I can provide you with advice for designing a board yourself (or by hiring someone else to do it for you), but can't really help much more than that.

Designing a version of my board that will feature at least Ethernet connectivity is something I myself wanted to do eventually anyway, but don't count on this happening any time soon.

**Comment by mrsstaceyseigle** [comment]
Quick question. Is there a way we can change the size (width) of the current board design you have without causing issues to the traces or will the traces adjust as well. I needed to shrink the board size down 3 mm

**Comment by magkopian** [comment]
@mrsstaceyseigle That depends, if after shrinking it you have traces outside the board outline there isn't much you can do about it, you need to move them. Instead of actually moving them though, it's probably easier to just delete the problematic traces and reroute them.

If you have no traces outside the board outline then just press `B` in order to redraw the ground planes, run a DRC check just to be sure and you should be good to go.

**Comment by mrsstaceyseigle** [comment]
Could you provide me with the outer dimensions (size) of this board

**Comment by magkopian** [comment]
@mrsstaceyseigle Sure, it's 93.25mm x 62.25mm.

**Comment by mrsstaceyseigle** [comment]
Do you have any suggestion of board manufactures to use that are able to make this pcb board. Fast turn times

**Comment by magkopian** [comment]
So, far I have experience with [ALLPCB](https://www.allpcb.com/), [JLCPCB](https://jlcpcb.com/) and [PCBWay](https://www.pcbway.com/). Don't have anything negative to say for any of them and generally they were all pretty fast each time I used them. For this board specifically I used ALLPCB and I was pretty happy with the result.

Only thing I'm not sure about is regarding the effect the recent outbreak had, as all these companies are based in China. It's been at least a couple of months since the last time I ordered PCBs from China so I can't tell you for certain that there won't be any delays.
